presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -962,6 +962,88 @@ private void testInsertPartitionedBucketedTableWithUnionAll(HiveStorageFormat st
     }
 
     @Test
+    public void testInsertBucketedPartition()
+            throws Exception
+    {
+        testInsertBucketedPartition(HiveStorageFormat.RCBINARY);
+    }
+
+    public void testInsertBucketedPartition(HiveStorageFormat storageFormat)
```

Comment:
There are two problems with this test:

* Your test would have passed with or without your change.
* This is an extremely round-about test, considering
  * It is in integration test for Hive (although Hive is not the only connector with bucket support)
  * It is intended to test planning, but end up go all the way to connector smoke test.

This [commit](https://github.com/prestodb/presto/commit/71a83937b2f9581a5eb858431a55b5dbe67d95b7) from you seems like a good example of testing code in AddExchange.